Lab 2 - Docker - Ahmed ElBatanony

ElBatanony opened the pull request.

safinsaf committed:
Add Lab1 task

Signed-off-by: safinsaf <safinsaft@gmail.com>

Cre-eD committed:
Update Lab1.md

safinsaf committed:
Fix minor typos

Signed-off-by: safinsaf <safinsaft@gmail.com>

safinsaf committed:
Remove creation of AWS account

Signed-off-by: safinsaf <safinsaft@gmail.com>

Commit:
Create the Moscow Time Python app

Commit:
Create tests for the Python app

Commit:
Add __pycache__ to the Python project gitignore

Commit:
Add .vscode to the repo's gitignore

Commit:
Edit markdown in Lab1.md to comply with linter

Commit:
Create the PYTHON.md file with best practices

Commit:
Add a README for the Python project

Commit:
Add a README for the repository

Commit:
Add the sample Flutter web project in app_flutter

Commit:
Add intl package to Flutter for date formatting

Commit:
Remove Flutter boilerplate and display Moscow Time

Commit:
Add FLUTTER.md with best practices

Commit:
Set the Flutter project details in the README

Commit:
Move markdown linter details to the repo README

Commit:
Move Python project details from PYTHON.md to README

Commit:
Add requirements.txt

Commit:
Add New BSD LICENSE

Commit:
Add DOCKER.md best practices

Commit:
Edit Flask app to run on host 0.0.0.0

Commit:
Add justification for choosing the Flask framework

Commit:
Add note on Docker linter to repo README

Commit:
Add Dockerfile for Flutter (simple HTTP server)

Commit:
Add justification for the Flutter web framework

Commit:
Add Dockerfile for Python Flask

Commit:
Add Docker section to Python README

Commit:
Add Docker section to Flutter README

safinsaf commented:
Good job, remove cache in upate dependencies step, update requirements to include only required packages, run image not as root user.

ElBatanony commented:
Thank you for the feedback! 😄  I applied the fixes in the lab 9 submission:
- Used the `--no-cache-dir` option with pip install
- Removed all packages from requirements.txt except for "Flask"
- Added a non-root user and used it in Dockerfile
